bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/DataFlowDiagramConverter.java

```diff
@@ -0,0 +1,403 @@
+
+                var assignment = ddFactory.createForwardingAssignment();
+                assignment.setOutputPin(outpin);
+                assignment.getInputPins().add(inpin);
+                behavior.getAssignment().add(assignment);
+            } else if (behaviorString.contains("set ")) {
+                String regex = "\\bset\\s+(\\S+)\\s+=\\s+(.+)\\b";
+
+                Pattern pattern = Pattern.compile(regex);
+                Matcher matcher = pattern.matcher(behaviorString);
+                if (matcher.find() && matcher.groupCount() == 2) {
+                    String variable = matcher.group(1);
+                    String typeName = variable.split("\\.")[0];
+                    String valueName = variable.split("\\.")[1];
+
+                    Label value = dd.getLabelTypes().stream().filter(labelType -> labelType.getEntityName().equals(typeName))
+                            .flatMap(labelType -> labelType.getLabel().stream()).filter(label -> label.getEntityName().equals(valueName)).findAny()
+                            .orElse(null);
+
+                    Assignment assignment = ddFactory.createAssignment();
+
+                    assignment.getInputPins().addAll(behavior.getInPin());
+                    assignment.setOutputPin(outpin);
+                    assignment.getOutputLabels().add(value);
+
+                    behavior.getAssignment().add(assignment);
+
+                    Term term = behaviorConverter.stringToTerm(matcher.group(2));
+
+                    assignment.setTerm(term);
+                }
+            }
+        }
+    }
+
+    private void putValue(Map<Node, Map<Pin, String>> nestedHashMap, Node node, Pin pin, String value) {
+        nestedHashMap.computeIfAbsent(node, k -> new HashMap<>()).put(pin, value);
+    }
+
+    public DataFlowDiagramAndDictionary webToDfd(String inputFile) {
+        return webToDfd(loadWeb(inputFile));
+    }
+
+    public DataFlowDiagramAndDictionary webToDfd(WebEditorDfd inputFile) {
+        return processWeb(inputFile);
+    }
+
+    public WebEditorDfd dfdToWeb(String inputDataFlowDiagram, String inputDataDictionary) {
+        DataFlowDiagramAndDictionary complete = loadDFD(inputDataFlowDiagram, inputDataDictionary);
+        return processDfd(complete.dataFlowDiagram(), complete.dataDictionary());
+    }
+
+    public WebEditorDfd dfdToWeb(DataFlowDiagramAndDictionary complete) {
+        return processDfd(complete.dataFlowDiagram(), complete.dataDictionary());
+    }
+
+    public void store(WebEditorDfd web, String outputFile) {
+        objectMapper = new ObjectMapper();
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+        try {
+            objectMapper.writeValue(new File(outputFile), web);
+        } catch (IOException e) {
+            logger.error(e);
+        }
+    }
+
+    public WebEditorDfd loadWeb(String inputFile) {
```

Comment:
As this Method either returns the value or null, what about using an optional here?